### PR TITLE
Add a class to aggregate container usage samples.

### PR DESCRIPTION
**Pull request by kgrygiel**
Introducing ContainerStats, a type which stores the aggregated usage history for a single container (CPU histogram + a sliding window with daily memory peaks from the last week). The history stored in this object will be populated from usage samples fetched from the real time usage data source with the AddSample() method.

The ContainerStats objects will be the main input for the Recommender to produce a recommendation for a container.

### CODE REVIEW

File: vertical-pod-autoscaler/recommender/util/histogram_mock.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// SubtractSample is a mock implementation of Histogram.SubtractSample.
+func (m *MockHistogram) SubtractSample(value float64, weight float64) {
+	m.Called(value, weight)
+}
+
+// IsEmpty is a mock implementation of Histogram.IsEmpty.
+func (m *MockHistogram) IsEmpty() bool {
+	args := m.Called()
+	return args.Bool(0)
+
```
Comment (mwielgus) [inline vertical-pod-autoscaler/recommender/util/histogram_mock.go]: empty line.

File: vertical-pod-autoscaler/recommender/model/container.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"time"
+
+	"k8s.io/autoscaler/vertical-pod-autoscaler/recommender/util"
+)
+
+// ContainerUsageSample is a measure of resource usage of a container over some
+// interval.
+type ContainerUsageSample struct {
+	// Start of the measurement interval.
+	measureStart time.Time
```
Comment (xiang90) [inline vertical-pod-autoscaler/recommender/model/container.go]: why not put end here explicitly?

Comment (xiang90) [inline vertical-pod-autoscaler/recommender/model/container.go]: never mind. the aggregation interval is a fixed const.

File: vertical-pod-autoscaler/recommender/util/histogram_mock.go (hunk 1 above)
Comment (kgrygiel) [inline vertical-pod-autoscaler/recommender/util/histogram_mock.go]: Done.

**Pull request merged**